src/client/app/public/login/login.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { UserService } from '../../core';
+import { DialogService } from '../../shared';
+
+@Component({
+  selector: 'login',
+  templateUrl: 'login.component.html',
+  styleUrls: ['login.component.scss']
+})
+
+export class LoginComponent implements OnInit {
+
+  public loginForm: FormGroup;
+
+  public constructor(
+    private userService: UserService,
+    private dialogService: DialogService,
+    private formBuilder: FormBuilder,
+    private router: Router
+  ) { }
+
+  public ngOnInit() {
+    this.buildLoginForm();
+  }
+
+  public buildLoginForm() {
+    this.loginForm = this.formBuilder.group({
+      email: ['', Validators.required],
+      password: ['', Validators.required]
+    });
+  }
+
+  public onClickLogin() {
+    this.login(this.loginForm.getRawValue());
+  }
+
+  private login(data: {email: string, password: string}) {
```

Comment:
This is getting a little bit dirty to look. Can we have the reference here to the entity?

Reply:
With that, I think it is better to remove the data. Only email and password.

Reply:
Yeah we can do that, I just want to accept an object if the data is more than one and related to each other. I think it should be an object, for unity to other functions and not to populate the parameters of a function.

Reply:
That's reasonable enough 👍